.github/workflows/client-build-lint-test.yml

```diff
@@ -0,0 +1,63 @@
+name: Client build, lint, and test
+on:
+    pull_request:
+        branches:
+            - dev
+        paths:
+            - 'app/client/**'
+
+jobs:
+    build:
+        name: Build
```

Comment:
The build step here is well, just building the project.  It doesn't actually do anything with the built files. All it really checks is that "can we build the project," which includes typechecking since in order to transpile the typescript to javascript the typescript compiler has to run.

There's not really a reason to wait for testing and linting before building. If the tests fail that doesn't necessarily mean the build fails and vice versa. They're all relatively independent.

Reply:
Makes sense, thanks for explaining your reasoning.